mllib/src/main/scala/org/apache/spark/ml/classification/ProbabilisticClassifier.scala

```diff
@@ -200,22 +200,17 @@ abstract class ProbabilisticClassificationModel[
     if (!isDefined(thresholds)) {
       probability.argmax
     } else {
-      val thresholds: Array[Double] = getThresholds
-      val probabilities = probability.toArray
+      val thresholds = getThresholds
       var argMax = 0
       var max = Double.NegativeInfinity
       var i = 0
       val probabilitySize = probability.size
       while (i < probabilitySize) {
-        if (thresholds(i) == 0.0) {
-          max = Double.PositiveInfinity
+        // thresholds are all > 0, excepting that at most one may be 0
+        val scaled = probability(i) / thresholds(i)
+        if (scaled > max) {
```

Comment:
You're right. It will never be predicted, and I think that's more sensible because probability = 0 means "never predict" and 0 threshold only sort of _implies_ always predicting the class. It's undefined, so either one seems coherent as a result. I prefer the current behavior I guess.

I see it's possible code-wise to have one class but don't think it's a valid use case, so, not worried about the behavior (even if it will still return the one single class here always anyway).